packages/sourcecred/src/core/ledger/ledger.js

```diff
@@ -819,23 +920,99 @@ export class Ledger {
     if (this._grainIntegrationStatuses.get(id) === true) {
       throw new Error("Integration has already executed this distribution");
     }
-    this._createAndProcessEvent({
-      type: "MARK_DISTRIBUTION_EXECUTED",
-      id,
-    });
+    this._grainIntegrationStatuses.set(id, true);
+  }
+
+  /**
+   *   Setting a currency is a prerequisite to executing a grain integration.
+   *
+   *   When accounting is enabled, this function will set an external currency
+   *   in the ledger, but will not modify any accounts.
+   *
+   *   When accounting is disabled, this function can be called to update
+   *   the ledger's external currency configuration. Note that this will also
+   *   deactivate identities missing a payout address that matches the new
+   *   configuration when accounting is disabled.
+   */
+  setExternalCurrency(chainId: ChainId, tokenAddress?: EthAddress): Ledger {
+    const currency = buildCurrency(chainId, tokenAddress);
+
+    if (!deepEqual(currency, this._externalCurrency)) {
+      this._createAndProcessEvent({
+        type: "SET_EXTERNAL_CURRENCY",
+        currency,
+      });
+      // Deactivate accounts if they don't have an address stored for
+      // the newly configured currency.
```

Comment:
nit: the code is sufficiently readable (yay verbose function naming!) so this comment seems redundant.